python/cudf/cudf/core/dataframe.py

```diff
@@ -350,18 +359,61 @@ def _setitem_tuple_arg(self, key, value):
                         )
                     else:
                         raise ValueError(
-                            f"shape mismatch: value array of shape "
-                            f"{value_df.shape} could not be "
-                            f"broadcast to indexing result of shape "
-                            f"{columns_df.shape}"
+                            error_msg.format(
+                                value1=value_df.shape,
+                                value2=columns_df.shape,
+                            )
                         )
                 else:
                     value_cols = value_df._data.columns
                 for i, col in enumerate(columns_df._column_names):
                     self._frame[col].loc[key[0]] = value_cols[i]
```

Comment:
If we are checking if `key[1]` is a scalar and do different thing below, we should probably do the same here.